### PR TITLE
Support YAML config format

### PR DESCRIPTION
**Pull request by gquintana**
This a mostly working prototype to be able to load YAML config (on par with JSON config) using https://github.com/FasterXML/jackson-dataformat-yaml

### CODE REVIEW

File: jmxtrans-core/src/main/java/com/googlecode/jmxtrans/guice/JmxTransModule.java [hunk 1]
```diff
@@ -83,7 +87,7 @@ protected void configure() {
 				.toInstance(getObjectPool(new SocketFactory(), SocketFactory.class.getSimpleName()));
 		bind(new TypeLiteral<GenericKeyedObjectPool<SocketAddress, DatagramSocket>>(){})
 				.toInstance(getObjectPool(new DatagramSocketFactory(), DatagramSocketFactory.class.getSimpleName()));
-		bind(KeyedObjectPool.class).annotatedWith(Names.named("mbeanPool"))
+		bind(new TypeLiteral<KeyedObjectPool<JmxConnectionProvider, JMXConnection>>(){}).annotatedWith(Names.named("mbeanPool"))
```
Comment (gehel) [inline jmxtrans-core/src/main/java/com/googlecode/jmxtrans/guice/JmxTransModule.java]: This change does not seem to be related to the title. Maybe split it in another PR?

Comment (gquintana) [inline jmxtrans-core/src/main/java/com/googlecode/jmxtrans/guice/JmxTransModule.java]: Well I don't know, but without that Guice complains that he can not find a bean of type ... and named mbeanPool, in short it doesn't work at all. Upgrading Jackson may have introduced some changes in dependencies.

File: jmxtrans-core/src/main/java/com/googlecode/jmxtrans/util/ProcessConfigUtils.java [hunk 2]
```diff
@@ -30,26 +30,34 @@
...
+		this.jsonMapper.setNodeFactory(placeholderResolverJsonNodeFactory);
+		this.yamlMapper.setNodeFactory(placeholderResolverJsonNodeFactory);
 	}
 
 	/**
 	 * Uses jackson to load json configuration from a File into a full object
 	 * tree representation of that json.
 	 */
 	public JmxProcess parseProcess(File file) throws IOException {
+		ObjectMapper mapper = file.getName().endsWith(".yml") || file.getName().endsWith(".yaml") ? yamlMapper : jsonMapper;
```
Comment (gehel) [inline jmxtrans-core/src/main/java/com/googlecode/jmxtrans/util/ProcessConfigUtils.java]: I would probably extract this line to its own method.

**Comment by gehel** [comment]
This looks good and simple enough. I was a bit worried of adding complexity to support multiple formats, but since this is simple enough, I have no objection. I suspect the failing integration test is a false positive, I restarted the build. If it fails again, I'll look deeper into it.

File: jmxtrans-core/src/main/java/com/googlecode/jmxtrans/guice/JmxTransModule.java (hunk 1 above)
Comment (gehel) [inline jmxtrans-core/src/main/java/com/googlecode/jmxtrans/guice/JmxTransModule.java]: Ok, fine by me.

**Comment by gquintana** [comment]
Should I change command line flags (--json-file --json-dir ) and environment vars (JSON_DIR) in start scripts?

**Comment by gehel** [comment]
Nope, don't change those. They are part of the external interface that users expect to be stable. If you really want to do the cleanup, introduce a new option and deprecate the old one (with a WARN log message).

**Comment by gquintana** [comment]
Documentation added https://github.com/jmxtrans/jmxtrans/wiki/YAMLConfig#using-standard-jsonyaml-configuration

**Comment by eesprit** [comment]
Sorry to ask it there, but it looked like the good place : Was it really merged in 266 like announced in the wiki page ?
Because I tested with a 266 build and it doesn't seem to recognise YAML at all.
Looking at the classes in the jar, there is no com/googlecode/jmxtrans/util/ProcessConfigUtils.class, and looking at the git repo for the 266 tag, the change is not there : https://github.com/jmxtrans/jmxtrans/tree/jmxtrans-parent-266/jmxtrans-core/src/main/java/com/googlecode/jmxtrans/util
Or Am I missing something ?
Are there some nighlty builds from master to test this ?

**Comment by gquintana** [comment]
@gehel What occured during release 266 https://travis-ci.org/jmxtrans/jmxtrans/builds/255802515 ?

**Comment by gehel** [comment]
This specific travis error is the release branch having be merged back to master before the travis build completed. That should not be a problem at all. To be honest, I have no idea what the issue was... I'll just release a new version and see if it works.
